packages/datadog-plugin-rhea/src/index.js

```diff
@@ -1,201 +1,79 @@
-            addToInFlightDeliveries(this.connection, msgObj.delivery)
-          }
-          return dispatch.apply(this, arguments)
-        })
-      }
+      this.enter(span, store)
+    })
 
-      return dispatch.apply(this, arguments)
-    }
-  }
-}
+    this.addSub(`apm:rhea:receive:start`, ({ msgObj, connection }) => {
+      const name = getResourceNameFromMessage(msgObj)
 
-function createWrapCircularBufferPopIf () {
-  return function wrapCircularBufferPopIf (popIf) {
-    return function wrappedPopIf (fn) {
-      arguments[0] = entry => {
-        const shouldPop = fn(entry)
-        if (shouldPop && entry[dd]) {
-          const remoteState = entry.remote_state
-          const state = remoteState && remoteState.constructor
-            ? entry.remote_state.constructor.composite_type : undefined
-          finish(entry, state)
-        }
-        return shouldPop
-      }
-      return popIf.apply(this, arguments)
-    }
-  }
-}
-
-function wrapDeliveryUpdate (update) {
-  return function wrappedUpdate (settled, stateData) {
-    if (this[dd]) {
-      const state = getStateFromData(stateData)
-      this[dd].span.setTag('amqp.delivery.state', state)
-    }
-    return update.apply(this, arguments)
-  }
-}
-
-function patchCircularBuffer (proto, instrumenter) {
-  Object.defineProperty(proto, 'outgoing', {
-    configurable: true,
-    get () {},
-    set (outgoing) {
-      delete proto.outgoing // removes the setter on the prototype
-      this.outgoing = outgoing // assigns on the instance, like normal
-      if (outgoing) {
-        let CircularBuffer
-        if (outgoing.deliveries) {
-          CircularBuffer = outgoing.deliveries.constructor
-        }
-        if (CircularBuffer && !patched.has(CircularBuffer.prototype)) {
-          instrumenter.wrap(CircularBuffer.prototype, 'pop_if', createWrapCircularBufferPopIf())
-          patched.add(CircularBuffer.prototype)
-          const Session = proto.constructor
-          if (Session) {
-            Session[circularBufferConstructor] = CircularBuffer
-          }
+      const store = storage.getStore()
+      const childOf = extractTextMap(msgObj, this.tracer)
+      const span = this.tracer.startSpan('amqp.receive', {
+        childOf,
+        tags: {
+          'span.type': 'worker',
+          'component': 'rhea',
+          'resource.name': name,
+          'service.name': this.config.service || this.tracer._service,
+          'span.kind': 'consumer',
+          'amqp.link.source.address': name,
+          'amqp.link.role': 'receiver'
         }
-      }
-    }
-  })
-}
+      })
+      analyticsSampler.sample(span, this.config.measured, true)
 
-function addToInFlightDeliveries (connection, delivery) {
-  let deliveries = connection[inFlightDeliveries]
-  if (!deliveries) {
-    deliveries = new Set()
-    connection[inFlightDeliveries] = deliveries
-  }
-  deliveries.add(delivery)
-  delivery[dd].connection = connection
```

Comment:
This appears to be the line that was lost in the refactor, leading to the Rhea memory leak.